### PR TITLE
speed up pixel DQM initialization by a factor >10

### PR DESCRIPTION
**Pull request by VinInn**
from >10 minutes to one minute
see issue #18606

need further cleanup (and verification!)

### CODE REVIEW

File: DQM/SiPixelPhase1Common/interface/SiPixelCoordinates.h [hunk 1]
```diff
@@ -171,27 +171,27 @@ class SiPixelCoordinates {
...
-  std::map<uint32_t, int> signed_disk_;
-  std::map<uint32_t, int> panel_;
-  std::map<uint32_t, int> ring_;
-  std::map<uint32_t, int> blade_;
-  std::map<uint32_t, int> signed_blade_;
-
-  std::map<uint32_t, unsigned int> fedid_;
-  std::map<uint64_t, unsigned int> channel_;
-  std::map<uint64_t, unsigned int> roc_;
+  std::unordered_map<uint32_t, int> quadrant_;
```
Comment (VinInn) [inline DQM/SiPixelPhase1Common/interface/SiPixelCoordinates.h]: all these maps HAVE to be filled at initialization, not on demand...

File: DQM/SiPixelPhase1Common/src/SiPixelCoordinates.cc [hunk 2]
```diff
@@ -35,6 +35,7 @@ void SiPixelCoordinates::init(edm::EventSetup const& iSetup) {
   edm::ESHandle<SiPixelFedCablingMap> cablingMapHandle;
   iSetup.get<SiPixelFedCablingMapRcd>().get(cablingMapHandle);
   cablingMap_ = cablingMapHandle.product();
+  fedid_ = cablingMap_->det2fedMap();
```
Comment (VinInn) [inline DQM/SiPixelPhase1Common/src/SiPixelCoordinates.cc]: fedid() method have to be cleanup...

File: CondFormats/SiPixelObjects/src/SiPixelFrameReverter.cc [hunk 3]
```diff
@@ -24,10 +24,10 @@ using namespace std;
 using namespace sipixelobjects;
 
 SiPixelFrameReverter::SiPixelFrameReverter(const edm::EventSetup& iSetup, const SiPixelFedCabling* map) 
-  : map_(map)
+  : map_(map), DetToFedMap(map->det2PathMap()) 
 { 
   // Build map
-  buildStructure(iSetup);
+  // buildStructure(iSetup);
```
Comment (VinInn) [inline CondFormats/SiPixelObjects/src/SiPixelFrameReverter.cc]: this is now obsolete, need to be cleanup.

**Comment by VinInn** [comment]
@cmsbuild, please test

**Comment by VinInn** [comment]
@jkarancs @veszpv

**Comment by VinInn** [comment]
I suggest to merge this (and even backport to 91X)
than cleanup the code

**Comment by ggovi** [comment]
+1

**Comment by dmitrijus** [comment]
+1
thanks!

**Comment by VinInn** [comment]
ping

**Comment by davidlange6** [comment]
indeed quite some cleanup is needed here - who is planning to do it? @fioriNTU

**Comment by fioriNTU** [comment]
@davidlange6 I will plan on the cleanup in this week, it is among our priorities

**Comment by franzoni** [comment]
+1

**Pull request merged**